### ISSUE TITLE
Does not work on Windows 7 Pro 64 bit

### ISSUE THREAD

**Issue by eb5agv** [acidhax/chromecast-audio-stream#18]
I can't get it to work on that platform. The Chromecast Audio is correctly configured, as it works from Chrome. But then when I select the device from the application icon and the right button, it is marked as selected but the icon does not change and there is no sound from the Chromecast Audio but from the standard PC speakers. Of course, I have rebooted all several times and there is no change.

**Comment by gsidhu** [comment]
I'm facing the exact same problem as eb5agv as well. Any fixes yet?

**Comment by krusse** [comment]
I'm using Windows 10 Pro x64, and the tray icon is completely unresponsive. Right click, left click, nothing happens. No device selection menu or anything. Works fine from chrome.

Edit: I moved the folder to C:\chromecast where before it was in a subfolder on my D drive. Now, I can select for my device as eb5agv above, but no chime, no sound also.

**Comment by arducode** [comment]
Hi, I'm another user with the same problem: I'm running Windows 7 64 bit and, even if the program sees my "Chromecast Audio" device and it appears checked, the sound is still played from my pc speakers and the chromecast doesn't seem to be connected to anything

**Comment by eduardorochabr** [comment]
I was able to fix this by downloading ffmpeg 64-bit. Just download the 64-bit version from the ffmpeg website, and then replace ffmpeg.exe from resources/bin/ffmpeg.

**Comment by ciscan81** [comment]
Hi eduardorochabr,

I'm having a similar problem with Windows 7 64-bit.

The Chromecast Audio Stream program runs and appears in my tray. I can connect to my Chromecast Audio and hear the 'connected' tone but cannot get any audio to stream from my desktop. In 'Volume Mixer' I cannot see ffmpeg.

I tried downloading and replacing ffmpeg.exe from resources/bin/ffmpeg as you described but no luck.

Which ffmpeg version did you download? 

Could you post a bit of a procedure?

Thanks

**Comment by eduardorochabr** [comment]
As a side note, on another machine with Windows 7 64-bit, it wasn't necessary to replace ffmpeg. 

The exact dir is "resources\bin\ffmpeg\win32", but I think you got it right.

The procedure is as straightforward as it can be, just extract either ffmpeg.exe and fprobe.exe into that folder after downloading the 64-bit version from the ffmpeg website.

The best way to debug the installation is running audio-cast.exe from the command line, can you do that and copy/paste the output here?

Run "cmd", then:

1) cd <chromecast dir>
2) audio-cast.exe

**Comment by lolouk44** [comment]
Hi.

I replaced ffmpeg and ffprobe with the 64bit versions, but it still doesn't work.
I get this when running from the command line:

```
Spawned Ffmpeg with command: ffmpeg -list_devices true -f dshow -i dummy pipe:1
An error occurred: ffmpeg exited with code 1: dummy: Immediate exit requested

C:\Program Files (x86)\Chromecast-Audio-Stream-1.0.0\resources\bin\driver\win32\
RegSvrEx.exe /c C:\Program Files (x86)\Chromecast-Audio-Stream-1.0.0\resources\b
in\driver\win32\audio_sniffer.dll
stdout:
stderr: 'C:\Program' is not recognized as an internal or external command,
operable program or batch file.

exec error: Error: Command failed: C:\Windows\system32\cmd.exe /s /c "C:\Program
 Files (x86)\Chromecast-Audio-Stream-1.0.0\resources\bin\driver\win32\RegSvrEx.e
xe /c C:\Program Files (x86)\Chromecast-Audio-Stream-1.0.0\resources\bin\driver\
win32\audio_sniffer.dll"
'C:\Program' is not recognized as an internal or external command,
operable program or batch file.
```

Any idea?

**Comment by eduardorochabr** [comment]
Move the folder "Chromecast-Audio-Stream-1.0.0" to C:\, then rename it.

So instead of:

C:\Program Files (x86)\Chromecast-Audio-Stream-1.0.0\

you get

C:\chromecast

**Comment by lolouk44** [comment]
Thanks,

Made the changes, and now I get this:

```
C:\chromecast
Example app listening at http://192.168.0.3:50655
Selected Audio Device:
Spawned Ffmpeg with command: ffmpeg -list_devices true -f dshow -i dummy pipe:1
An error occurred: ffmpeg exited with code 1: dummy: Immediate exit requested

C:\chromecast\resources\bin\driver\win32\RegSvrEx.exe /c C:\chromecast\resources
\bin\driver\win32\audio_sniffer.dll
stdout: hello there
stderr:
Spawned Ffmpeg with command: ffmpeg -list_devices true -f dshow -i dummy pipe:1
An error occurred: ffmpeg exited with code 1: dummy: Immediate exit requested
```

and from the time I select a device to cast to:

```
ffmpeg version N-79690-g78baa45 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 5.3.0 (GCC)
  configuration: --enable-gpl --enable-version3 --disable-w32threads --enable-av
isynth --enable-bzlib --enable-fontconfig --enable-frei0r --enable-gnutls --enab
le-iconv --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --
enable-libfreetype --enable-libgme --enable-libgsm --enable-libilbc --enable-lib
modplug --enable-libmfx --enable-libmp3lame --enable-libopencore-amrnb --enable-
libopencore-amrwb --enable-libopenjpeg --enable-libopus --enable-librtmp --enabl
e-libschroedinger --enable-libsnappy --enable-libsoxr --enable-libspeex --enable
-libtheora --enable-libtwolame --enable-libvidstab --enable-libvo-amrwbenc --ena

ble-libvorbis --enable-libvpx --enable-libwavpack --enable-libwebp --enable-libx
264 --enable-libx265 --enable-libxavs --enable-libxvid --enable-libzimg --enable
-lzma --enable-decklink --enable-zlib
  libavutil      55. 23.100 / 55. 23.100
  libavcodec     57. 38.100 / 57. 38.100
  libavformat    57. 34.103 / 57. 34.103
  libavdevice    57.  0.101 / 57.  0.101
  libavfilter     6. 44.100 /  6. 44.100
  libswscale      4.  1.100 /  4.  1.100
  libswresample   2.  0.101 /  2.  0.101
  libpostproc    54.  0.100 / 54.  0.100
[dshow @ 0000000000399300] Could not find audio only device with name [virtual-a
udio-capturer] among source devices of type audio.
[dshow @ 0000000000399300] Searching for audio device within video devices for v
irtual-audio-capturer
[dshow @ 0000000000399300] Could not find audio only device with name [virtual-a
udio-capturer] among source devices of type video.
audio=virtual-audio-capturer: I/O error

An error occurred: ffmpeg exited with code 1: dummy: Immediate exit requested

C:\chromecast\resources\bin\driver\win32\RegSvrEx.exe /c C:\chromecast\resources
\bin\driver\win32\audio_sniffer.dll
stdout: hello there
stderr:
Spawned Ffmpeg with command: ffmpeg -list_devices true -f dshow -i dummy pipe:1
An error occurred: ffmpeg exited with code 1: dummy: Immediate exit requested
```

**Comment by eduardorochabr** [comment]
The 64-bit ffmpeg didn't work for you.

Go back to the 32-bit ffmpeg executables and try again from the command line. Paste here.

Also after getting the 32-bit ffmpeg, execute this command and paste here:

c:\chromecast\resources\bin\ffmpeg\win32\ffmpeg -list_devices true -f dshow -i dummy pipe:1

**Comment by lolouk44** [comment]
Thanks, it's working now :)

**Comment by ciscan81** [comment]
Hi eduardorochabr

I followed your instructions too regarding placement of the 'Chromecast' folder directly in C:\

Didn't work with the 64bit FFmpeg files. I replaced them with the 32bit versions and now I'm merrily Chromecasting to my Stereo as I write : )

I guess placement of the folder is critical somehow?

Thanks again for your help.

**Comment by eduardorochabr** [comment]
Hi ciscan81,

I have an installation of Windows 7 64-bit that also didn't work with the
64-bit ffmpeg.

I think most installs won't need to bother with replacing the original
32-bit ffmpeg.

However, for one install I really had to replace them to make it work, so
it's nice to have this documented here in this thread as a possible
workaround.

Thanks for the report!

2016-05-05 21:16 GMT-03:00 ciscan81 notifications@github.com:

**Comment by SlavikCA** [comment]
Tried to run it on my Win 7 x64 and it doesn't work.
Here is what I got:

```
c:\cast\resources\bin\ffmpeg\win32>ffmpeg -list_devices true -f dshow -i dummy pipe:1
ffmpeg version N-69406-ga73c411 Copyright (c) 2000-2015 the FFmpeg developers
  built on Jan 29 2015 23:07:55 with gcc 4.9.2 (GCC)
  configuration: --arch=x86 --target-os=mingw32 --cross-prefix=/home/videostream/Downloads/sandbox/mingw-w64-i686/bin/i686-w64-mingw32- --pkg-config=pkg-config --enable-gpl --enable-libx264 --enable-avisynth --enable-libxvid --enable-libmp3lame --enable-version3 --enable-zlib --enable-librtmp --enable-libvorbis --enable-libtheora --enable-libspeex --enable-libopenjpeg --enable-gnutls --enable-libgsm --enable-libfreetype --enable-libopus --disable-w32threads --enable-frei0r --enable-filter=frei0r --enable-libvo-aacenc --enable-bzlib --enable-libxavs --extra-cflags=-DPTW32_STATIC_LIB --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libvo-amrwbenc --enable-libschroedinger --enable-libvpx --enable-libilbc --prefix=/home/videostream/Downloads/sandbox/mingw-w64-i686/i686-w64-mingw32 --enable-static --disable-shared --enable-libsoxr --enable-fontconfig --enable-libass --enable-libutvideo --enable-libbluray --enable-iconv --enable-libtwolame --extra-cflags=-DLIBTWOLAME_STATIC --enable-libzvbi --enable-libcaca --enable-libmodplug --extra-libs=-lstdc++ --extra-libs=-lpng --enable-libvidstab --enable-libx265 --enable-decklink --extra-libs=-loleaut32 --extra-cflags= --extra-cflags= --enable-nonfree --enable-libfdk-aac --disable-libfaac --disable-decoder=aac --enable-runtime-cpudetect
  libavutil      54. 18.100 / 54. 18.100
  libavcodec     56. 21.101 / 56. 21.101
  libavformat    56. 19.100 / 56. 19.100
  libavdevice    56.  4.100 / 56.  4.100
  libavfilter     5.  9.101 /  5.  9.101
  libswscale      3.  1.101 /  3.  1.101
  libswresample   1.  1.100 /  1.  1.100
  libpostproc    53.  3.100 / 53.  3.100
[dshow @ 049a1ae0] DirectShow video devices (some may be both video and audio devices)
[dshow @ 049a1ae0]  "Integrated Webcam"
[dshow @ 049a1ae0]     Alternative name "@device_pnp_\\?\usb#vid_0c45&pid_6449&mi_00#7&27cf33fe&0&0000#{65e8773d-8f56-11d0-a3b9-00a0c9223196}\global"
[dshow @ 049a1ae0] DirectShow audio devices
[dshow @ 049a1ae0]  "Microphone Array (IDT High Defi"
[dshow @ 049a1ae0]     Alternative name "@device_cm_{33D9A762-90C8-11D0-BD43-00A0C911CE86}\Microphone Array (IDT High Defi"
[dshow @ 049a1ae0]  "virtual-audio-capturer"
[dshow @ 049a1ae0]     Alternative name "@device_sw_{33D9A762-90C8-11D0-BD43-00A0C911CE86}\{8E14549B-DB61-4309-AFA1-3578E927E935}"
[dshow @ 049a1ae0]  "Microphone (NoMachine Microphon"
[dshow @ 049a1ae0]     Alternative name "@device_cm_{33D9A762-90C8-11D0-BD43-00A0C911CE86}\Microphone (NoMachine Microphon"
[dshow @ 049a1ae0]  "Rec. Playback (IDT High Definit"
[dshow @ 049a1ae0]     Alternative name "@device_cm_{33D9A762-90C8-11D0-BD43-00A0C911CE86}\Rec. Playback (IDT High Definit"
dummy: Immediate exit requested

```

**Comment by norsemangrey** [comment]
I have the same issue. I try running the application a Win 7 x64 computer. I get the icon and can connect to the Chromecast device (I get the connected sound), but the icon does not change and it will not stream any audio.

I have tried moving the application folder to C:/ and renaming it to Chromecast. I have also tried changing the ffmpeg file to an x64 version without success so I changed it back again to the 32 bit version.

Any more tips on how to resolve this issue? I would really low to be able to stream audio from my computer :)
